src/Castle.Core/DynamicProxy/Generators/BaseProxyGenerator.cs

```diff
@@ -38,14 +38,19 @@ namespace Castle.DynamicProxy.Generators
 	internal abstract class BaseProxyGenerator
 	{
 		protected readonly Type targetType;
+		protected readonly Type[] interfaces;
 		private readonly ModuleScope scope;
 		private ILogger logger = NullLogger.Instance;
 		private ProxyGenerationOptions proxyGenerationOptions;
 
-		protected BaseProxyGenerator(ModuleScope scope, Type targetType, ProxyGenerationOptions proxyGenerationOptions)
+		protected BaseProxyGenerator(ModuleScope scope, Type targetType, Type[] interfaces, ProxyGenerationOptions proxyGenerationOptions)
```

Comment:
Additional interfaces are allowed for all proxy types, so it makes sense that the base class asks for them.